sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -551,6 +586,504 @@ Parameters:
+executable graphs created from that modifiable graph until `command_graph::update()`
+is called passing the modified nodes, or a new executable graph is finalized from
+the modifiable graph.
+
+It is not an error if `newBufferRef` is set to the current parameter value in any
+registered nodes.
+
+Parameters:
+
+* `newBufferRef` - The new buffer that the dynamic accessor will provide access to
+in all registered graph nodes.
+* `newAccessRange` - Access range for the accessor of the new buffer parameter.
+
+Exceptions:
+
+Throws an exception with the errc::invalid error code if newAccessRange exceeds
+the range of bufferRef in any dimension.
+
+|
+[source,c++]
+----
+template <typename AllocatorT>
+void update(
+        buffer<DataT, Dimensions, AllocatorT> &newBufferRef,
+        range<Dimensions> newAccessRange,
+        id<Dimensions> newAccessOffset);
+----
+
+|Updates the buffer that this dynamic accessor provides access to in all graph nodes
+where it is registered. This new value will be reflected immediately in the modifiable
+graph which contains the registered nodes. The new value will not be reflected in any
+executable graphs created from that modifiable graph until `command_graph::update()`
+is called passing the modified nodes, or a new executable graph is finalized from
+the modifiable graph.
+
+It is not an error if `newBufferRef` is set to the current parameter value in any
+registered nodes.
+
+Parameters:
+
+* `newBufferRef` - The new buffer that the dynamic accessor will provide access to
+in all registered graph nodes.
+* `newAccessRange` - Access range for the accessor of the new buffer parameter.
+* `newAccessOffset` - Access offset for the range of the new buffer parameter.
+
+Exceptions:
+
+Throws an exception with the errc::invalid error code if the sum of
+newAccessRange and newAccessOffset exceeds the range of bufferRef in any dimension.
+
+
+|===
+
+===== The dynamic_local_accessor class [[dynamic-local-accessor-class]]
+
+[source,c++]
+----
+namespace ext::oneapi::experimental{
+template <typename DataT, int Dimensions = 1>
+class dynamic_local_accessor {
+public:
+  dynamic_local_accessor(
+        command_graph<graph_state::modifiable> graph,
+        const property_list &propList = {});
+
+  dynamic_local_accessor(
+        command_graph<graph_state::modifiable> graph,
+        range<Dimensions> allocationSize,
+        const property_list &propList = {});
+
+  void update(range<Dimensions> newAllocationSize);
+};
+}
+----
+
+The `dynamic_local_accessor` class provides the {crs}[common reference semantics].
+
+Table {counter: tableNumber}. Member functions of the `dynamic_local_accessor` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+dynamic_local_accessor(
+      command_graph<graph_state::modifiable> graph,
+      const property_list &propList = {});
+----
+|Available only when (Dimensions == 0).
```

Comment:
Will probably remove this constructor. Even though it exists for local_accessors, it doesn't seem to make sense for dynamic_local_accessors since a 0 dimension local accessors, by definition, has size == 1 which cannot be modified.